### PR TITLE
Fixes #31119 - run katello-certs-check when installing with custom certs

### PR DESCRIPTION
**Pull request by wbclark**
Based on #514

### CODE REVIEW

**Comment by ehelms** [comment]
I think you can rebase this now @wbclark

**Comment by wbclark** [comment]
Rebased and ready for review

**Comment by ekohl** [comment]
Looks like the tests don't like the expired certificate, other than that I think this makes sense.

Edit: perhaps this also deserves a Redmine issue for the changelog?

**Review by ekohl — CHANGES_REQUESTED** [review]
Please open a Redmine issue for this and use a more descriptive commit message, other than that :+1:

File: hooks/pre_commit/20-certs_update.rb [hunk 1]
```diff
@@ -13,8 +13,8 @@
   cert_file = param('certs', 'server_cert').value
   key_file  = param('certs', 'server_key').value
 
-  if !app_value(:certs_skip_check) && !cert_file.to_s.empty? &&
-     (app_value(:certs_update_server_ca) || app_value(:certs_update_server))
+  if !app_value(:certs_skip_check) && (!cert_file.to_s.empty? ||
+      !ca_file.to_s.empty? || !key_file.to_s.empty?)
```
Comment (ekohl) [inline hooks/pre_commit/20-certs_update.rb]: Would this be easier to understand?
```suggestion
  unless app_value(:certs_skip_check) || cert_file.to_s.empty? || ca_file.to_s.empty? || key_file.to_s.empty?
```

Alternative:
```suggestion
  unless app_value(:certs_skip_check) || [cert_file, ca_file, key_file].any? { |v| v.to_s.empty? }
```

Comment (wbclark) [reply hooks/pre_commit/20-certs_update.rb]: I think the logic is not correct. Converting it step by step:

1. `if !app_value(:certs_skip_check) && (!cert_file.to_s.empty? || !ca_file.to_s.empty? || !key_file.to_s.empty?)`

2. `if !app_value(:certs_skip_check) && !(cert_file.to_s.empty? && ca_file.to_s.empty? && key_file.to_s.empty?)`

3. `unless app_value(:certs_skip_check) || (cert_file.to_s.empty? && ca_file.to_s.empty? && key_file.to_s.empty?)`

4. `unless app_value(:certs_skip_check) || [cert_file, ca_file, key_file].all? { |v| v.to_s.empty? }`

I do rather like that and agree it's easier to read. I'll update it in a moment. Thanks for the suggestion!

**Review by ekohl — APPROVED** [review]

**Pull request merged**